commons/c2cgeoportal_commons/alembic/static/44769e643e2f_add_technical_users.py

```diff
@@ -0,0 +1,107 @@
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2019, Camptocamp SA
+# All rights reserved.
+
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+
+# 1. Redistributions of source code must retain the above copyright notice, this
+#    list of conditions and the following disclaimer.
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions and the following disclaimer in the documentation
+#    and/or other materials provided with the distribution.
+
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+# ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+# ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+# (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+# ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+# SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+# The views and conclusions contained in the software and documentation are those
+# of the authors and should not be interpreted as representing official policies,
+# either expressed or implied, of the FreeBSD Project.
+
+"""Add technical users
+
+Revision ID: 44769e643e2f
+Revises: bd029dbfc11a
+Create Date: 2019-10-02 16:13:29.001061
+"""
+
+from alembic import op
+from c2c.template.config import config
+from sqlalchemy import Column, MetaData, Table
+from sqlalchemy.types import Unicode, String
+
+# revision identifiers, used by Alembic.
+revision = '44769e643e2f'
+down_revision = 'bd029dbfc11a'
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    schema = config['schema']
+    staticschema = config['schema_static']
+
+    user = Table(
+        'user', MetaData(),
+        Column('username', Unicode),
+        Column('type', String(10), nullable=False),
+        Column('password', Unicode, nullable=False),
+        Column('email', Unicode, nullable=False),
+        schema=staticschema,
+    )
+
+    op.bulk_insert(user, [
+        {'username': name, 'type': 'user', 'password': 'x', 'email': '{}@example.com'.format(name)}
+        for name in ('anonymous', 'registered', 'intranet')
+    ])
```

Comment:
Why do we need to create users, as far as I know functionnalities are associated to roles.
In case we really need thems, be careful that these abstract users cannot log in.

Reply:
I start to be base on user, but it's true that nor needed, except if we want to change the groups attached to the user. Technically it's true that not needed.
What do you think @jwkaltz , @ybolognini ?

Reply:
I don't like the technical users, we should avoid them if possible